n3fit/src/evolven3fit_new/eko_utils.py

```diff
@@ -58,46 +57,74 @@ def construct_eko_cards(
     theory = Loader().check_theoryID(theoryID).get_description()
     theory.pop("FNS")
     theory.update(theory_card_dict)
-    if "nfref" not in theory:
-        theory["nfref"] = NFREF_DEFAULT
-    if "nf0" not in theory:
-        theory["nf0"] = NF0_DEFAULT
-        
+
     # Prepare the thresholds according to MaxNfPdf
     thresholds = {"c": theory["kcThr"], "b": theory["kbThr"], "t": theory["ktThr"]}
     if theory["MaxNfPdf"] < 5:
         thresholds["b"] = np.inf
     if theory["MaxNfPdf"] < 6:
         thresholds["t"] = np.inf
-        
+
+    if "nfref" not in theory:
+        theory["nfref"] = NFREF_DEFAULT
+
+    # Set nf_0 according to the fitting scale unless set explicitly
+    mu0 = theory["Q0"]
+    if "nf0" not in theory:
+        if mu0 < theory["mc"] * thresholds["c"]:
+            theory["nf0"] = 3
+        elif mu0 < theory["mb"] * thresholds["b"]:
+            theory["nf0"] = 4
+        elif mu0 < theory["mt"] * thresholds["t"]:
+            theory["nf0"] = 5
+        else:
+            theory["nf0"] = 6
```

Comment:
this should be the same as `int(nf_default(mu0**2, atlas))` with the `atlas` from below (you can first use an atlas with a random origin)

PS: @AleCandido you see: for outsiders we need to preserve `nf_default`

Reply:
If I need to create a fake atlas I prefer to have this explicit here. All things considered it is not very ugly and it makes very explicit what nf0 means.

Reply:
In which case `nf0` will not be present in the theory card? Maybe the old ones?

Reply:
In the nnpdf theory database it's not present (although the information is there because we have mu0 and the thresholds.